compiler/definitions/ir/file_id.py

```diff
@@ -118,3 +121,13 @@ def isNull(self):
 
     def get_ident(self):
         return self.ident
+
+    def is_available_on(self, host):
+        if self.is_ephemeral():
+            return True
+        elif self.has_remote_file_resource():
+            return self.resource.is_available_on(host)
+        else:
+            # Currently any other resource types should
```

Comment:
And if I understand correctly, the main graph never calls `is_available_on`?

Reply:
yeah, this function is for evaluating if a fid can be serviced on the provided host. The "main graph" by design runs in the user's shell so we should never call is_available_on on the main graph fids.

The calls for this function are in worker_manager.py

Reply:
cool!